king_phisher/client/windows/campaign_import.py

```diff
@@ -0,0 +1,439 @@
+		self.button_import_campaign.set_sensitive(False)
+		self.select_button = self.gobjects['button_select']
+		self.entry_campaign_name = self.gobjects['entry_campaign']
+		self.entry_path = self.gobjects['entry_file']
+		self.import_progress = self.gobjects['progressbar']
+		self.spinner = self.gobjects['spinner']
+		self.text_buffer = self.gobjects['textview'].get_buffer()
+
+		# place holders for once an xml file is loaded
+		self.campaign_info = None
+		self.db_campaigns = None
+		self.thread_import_campaign = None
+
+	def _set_text_view(self, string_to_set):
+		self.text_buffer.set_text(string_to_set + "\n")
+
+	def _update_text_view(self, string_to_add):
+		end_iter = self.text_buffer.get_end_iter()
+		self.text_buffer.insert(end_iter, string_to_add + "\n")
+
+	def _update_id(self, element, id_fields, old_id, new_id):
+		"""
+		Iterates through the element and replaces the specified old ID to the new ID in the requested ID fields.
+
+		:param element: Element to iterate over where the old id values can be found.
+		:type element: :py:class:`xml.etree.ElementTree.Element`
+		:param list id_fields: The list of fields to look for old_id.
+		:param old_id: The old id value that has been changed
+		:param new_id: The new id value to set.
+		"""
+		for nods in element.iter():
+			if nods.tag in id_fields and nods.text == old_id:
+				nods.text = new_id
+				# if new_id is none set type to null
+				if new_id is None:
+					nods.attrib['type'] = 'null'
+
+	def _get_keys_values(self, element):
+		keys = []
+		rows = []
+		for subelements in element:
+			values = []
+			for node in subelements:
+				keys.append(node.tag) if node.tag not in keys else None
+				if node.attrib['type'] in ('datetime', 'date', 'time'):
+					values.append(serializers.from_elementtree_element(node).replace(tzinfo=None))
+				else:
+					values.append(serializers.from_elementtree_element(node))
+			rows.append(tuple(values))
+		return keys, rows
+
+	def interact(self):
+		self.window.set_position(Gtk.WindowPosition.CENTER_ON_PARENT)
+		self.window.show_all()
+
+	def entry_change(self, _):
+		"""
+		When there is a change in the campaign entry field it will check to see if the name is already in use.
+		If it is not in use it will change the Sensitivity of the Import Campaign Button to allow the user to
+		start the import process.
+		"""
+		if not self.campaign_info:
+			return
+		if not self._check_campaign_name(self.entry_campaign_name.get_text()):
+			self.button_import_campaign.set_sensitive(False)
+			return
+		self.button_import_campaign.set_sensitive(True)
+		return
+
+	def open_xml_file(self, _):
+		if not self.select_button.get_property('sensitive'):
+			return
+		self.select_xml_campaign()
+
+	def close_import_campaign(self, _):
+		"""
+		Checks to make sure the import campaign thread is closed before closing.
+		"""
+		if not self.campaign_info:
+			return
+		if not self.thread_import_campaign:
+			return
+		if not self.thread_import_campaign.isAlive():
```

Comment:
Use the newer and pep8 compliant name of `.is_alive` here instead of the camelCase one.